SetupDataPkg/ConfApp/ConfApp.c

```diff
@@ -86,11 +89,47 @@ CONST ConfAppKeyOptions  MainStateOptions[MAIN_STATE_OPTIONS] = {
 ConfState_t                        mConfState       = MainInit;
 EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL  *mSimpleTextInEx = NULL;
 DFCI_SETTING_ACCESS_PROTOCOL       *mSettingAccess  = NULL;
+BOOLEAN                            mInitialized     = FALSE;
 DFCI_AUTH_TOKEN                    mAuthToken;
 DFCI_IDENTITY_MASK                 mIdMask;          // Identities installed
 SECURE_BOOT_PAYLOAD_INFO           *mSecureBootKeys;
 UINT8                              mSecureBootKeysCount;
 
+/**
+  Quick helper function to see if ReadyToBoot has already been signalled.
+
+  @retval     TRUE    ReadyToBoot has been signalled.
+  @retval     FALSE   Otherwise...
+
+**/
+BOOLEAN
+IsPostReadyToBoot (
+  VOID
+  )
+{
+  EFI_STATUS       Status;
+  UINT32           Attributes;
+  PHASE_INDICATOR  Indicator;
+  UINTN            Size;
+  static BOOLEAN   Result = FALSE;

```

Comment:
does this need to be static? Or can this be removed in favor of using mInitialized differently?

Reply:
It is removed. Since we only query this at the beginning of 2 pages, using cached mechanism is not that beneficial. So it is changed to query variable state in case some other in-app operations will trigger ready to boot and caused us in a torn state.